crates/buffered_watch/src/receiver.rs

```diff
@@ -0,0 +1,150 @@
+use std::{cell::UnsafeCell, ops::Deref, sync::Arc};
+
+use parking_lot::{RwLock, RwLockReadGuard};
+use tokio::sync::watch;
+
+use crate::{find_newest_readable_buffer, find_oldest_free_buffer, NoSender, Shared, State};
+
+/// Receives values from the associated Sender
+pub struct Receiver<T> {
+    pub(crate) shared: Arc<RwLock<Shared<T>>>,
+    pub(crate) notifier: watch::Receiver<()>,
+}
+
+unsafe impl<T> Sync for Receiver<T> where T: Sync {}
+unsafe impl<T> Send for Receiver<T> where T: Send + Sync {}
+
+impl<T> Receiver<T> {
+    /// Waits for the next change and marks the buffer as seen
+    pub async fn wait_for_change(&mut self) -> Result<(), NoSender> {
+        self.notifier.changed().await.map_err(|_| NoSender)
+    }
```

Comment:
I am not sure whether I got your suggestion.

We (@knoellle and I) are sure that the borrow and the mark unchanged **must** be handled together. Otherwise we loose updates or send them twice.

So if you suggested to move the `mark_unchanged()` to somewhere else than the borrow call, this cannot work.

Reply:
I think, we would not loose updates, but may get them twice. And as I said, this would not matter much. But either way, it should not harm correctness, which also means, your solution works, but needs two `borrow*()` methods. Feel free to resolve if read.